src/app/modules/components/courses/add-edit-form/add-edit-form.component.ts

```diff
@@ -13,12 +15,16 @@ import { FormBuilder } from '@angular/forms';
 export class AddEditFormComponent {
 
   constructor(
-    @Inject(MAT_DIALOG_DATA) public courseToEdit: Course,
+    @Inject(MAT_DIALOG_DATA) public courseToEditId: number,
     private coursesService: CoursesService,
-    private formBuilder: FormBuilder) { }
+    private formBuilder: FormBuilder,
+    public dialogRef: MatDialogRef<AddEditPageComponent>,
+    ) {
+    console.log(this.courseToEditId, this.coursesService.getCourseById(this.courseToEditId));
+  }
 
-  public course: Course = this.courseToEdit || {title: '', description: '', creationDate: '', duration: 0};
-  public isEdit: Boolean = !!this.courseToEdit;
+  public course: Course = this.coursesService.getCourseById(this.courseToEditId) || {id: 0, title: '', description: '', creationDate: '', duration: 0};
```

Comment:
It is not quite good practice to define such empty object as default value. it is much better to set null